fixed endless loading of collab tab and redirect to /tz problem

xat opened the pull request:
this fixes the problem mentioned in https://github.com/teia-community/teia-ui/issues/2 and a further problem that triggered a redirect to '/tz' when clicking on the collab tab as a user without subjkt.

xat committed:
fixed endless loading of collab tab and redirect to /tz problem

melMass commented on src/pages/display/index.js:
```diff
@@ -579,12 +579,11 @@ export default class Display extends Component {
-    const { subjkt, wallet } = this.state
-    this.props.history.push(`/${subjkt === '' ? wallet : subjkt}/${slug}`)
+    if (this.state.subjkt !== '') {
+      this.props.history.push(`/${this.state.subjkt}/collabs`)
```
Can't we just always resolve to the address?

xat replied:
hmm, it's done like this in the other tabs as well and ensures that the url contains the username.

melMass replied:
Ah my bad, I thought routing handled these

LeonNicholls reviewed (approved):
LGTM

melMass reviewed (commented):
@xat Very nice, do you think it's out of scope of this issue to either dim the tab title (or hide?), or show No Collaborations in the "grid view" ?

melMass committed:
feat: ✨ add "No Collabs" indication

melMass commented:
@xat I made a simple proposal in 6151985157c5c1889037acfe6e7b930f55a25c2a

![image](https://user-images.githubusercontent.com/7041726/159556061-2929a5a0-e69f-47b2-9c6b-bbb1d2cea033.png)

melMass committed:
fix: 🎨 Added collections to the proposal

melMass commented:
Also added collections, discard loading components if the objkts length === 0

![image](https://user-images.githubusercontent.com/7041726/159555949-568860e2-c22e-4d4f-a988-bc2d488b6594.png)

melMass commented:
This might more be out of scope for this issue,
but the first render of an artist page is strange, to reproduce, go to the feed: https://teia-ui-pr-32.onrender.com/
Click on an artist name down an objkt, there! something strange happens on "route resolution"
The name is already resolved so it should be used as the path when there is a SUBJKT

melMass commented:
(rebased)

xat commented on src/pages/display/index.js:
```diff
@@ -1062,151 +1065,166 @@ export default class Display extends Component {
-
-        {!this.state.loading && this.state.collectionState && (
-          <div>
+        {((this.state.collabsState && this.state.collabs.length === 0) ||
```
collabs length is always 0 because they are loaded in the CollabsTab component. this means currently always 'No Collabs' is shown.

<img width="953" alt="Bildschirmfoto 2022-03-23 um 08 09 11" src="https://user-images.githubusercontent.com/182512/159643580-96193e47-406e-4958-a470-3a49501e0576.png">

xat replied:
fixed

xat committed:
added empty-state to creations tab as well, small refactorings, moved empty-state of collabs-tab into CollabsTab component

xat commented:
did some small refactorings on the tab logic, moved the empty-state of the collabs tab into the CollabsTab component, and added the empty-state to the creations tab as well.

this means another round of testing is needed :-)

i think this whole component needs heavy refactoring and would greatly benefit from a library like `useSWR`.

yes, the routing is somehow odd and also needs refactoring. the link to the artist subpage from the start-page should directly point to the slug-version of the artist profile if the artist has a slug, so that these redirects are not needed in the first place. but imo fixing that is out of scope for this pr.

melMass commented:
> i think this whole component needs heavy refactoring and would greatly benefit from a library like `useSWR`.

Yes my first commit I realised I did not understand it was a huge "if" list 😅 .....
Thanks for the fix! Another thing I've realised is that we can also have `loaded = false` & `collection.length==0` for a second when landing on the tab. There might be a more clever way to avoid that...

melMass reviewed (changes requested):
Looks good !

I just have the dead road issue resulting in a blank profile with "no creation":
https://teia-ui-pr-32.onrender.com/ROBNESS_V3

Should'nt we have a 404?

xat commented:
yes, that's a problem since forever. see for example https://www.hicetnunc.xyz/ROBNESS_V3

do we want to tackle it as part of this pr? otherwise, i would suggest that one of the designers comes up with a nice TEIA-404 image and we implement that in a separate issue.

xat commented:
> > i think this whole component needs heavy refactoring and would greatly benefit from a library like `useSWR`.
> 
> Yes my first commit I realised I did not understand it was a huge "if" list 😅 ..... Thanks for the fix! Another thing I've realised is that we can also have `loaded = false` & `collection.length==0` for a second when landing on the tab. There might be a more clever way to avoid that...

`useSWR` would be the more clever way, but a complete refactor of the component would be needed :-)

melMass commented:
> i would suggest that one of the designers comes up with a nice TEIA-404 image and we implement that in a separate issue.

Yep agreed! Let's open a dedicated issue for other devs to pickup ;)
